marklogic-langchain4j/src/main/java/com/marklogic/langchain4j/splitter/JsonChunkDocumentProducer.java

```diff
@@ -84,18 +97,12 @@ private String determineChunksArrayName(ObjectNode doc) {
         return doc.has(DEFAULT_CHUNKS_ARRAY_NAME) ? "splitter-chunks" : DEFAULT_CHUNKS_ARRAY_NAME;
     }
 
-    private ArrayNode generateConceptsForChunk(String text) {
-        Map<String, Collection<ClassificationScore>> classificationScores = chunkConfig.getClassifier().classifyText(super.sourceDocument.getUri(), text);
-        ArrayNode conceptsArray = objectMapper.createArrayNode();
-        for (Map.Entry<String, Collection<ClassificationScore>> entry : classificationScores.entrySet()) {
-            for (ClassificationScore classificationScore : classificationScores.get(entry.getKey())) {
-                ObjectNode concept = conceptsArray.addObject();
-                concept.put("id", classificationScore.getId());
-                concept.put("name", entry.getKey());
-                concept.put("value", classificationScore.getName());
-                concept.put("score", classificationScore.getScore());
-            }
+    private JsonNode generateClassificationForChunk(String text) {
+        byte[] result = chunkConfig.getClassifier().classifyTextToBytes(super.sourceDocument.getUri(), text);
+        try {
+            return xmlMapper.readTree(result);
+        } catch (IOException e) {
+            throw new MarkLogicLangchainException(String.format("Unable to generate JSON classification; cause: %s", e.getMessage()), e);
```

Comment:
Include the sourceUri in this error message too; otherwise, a connector/Flux user won't have any idea what record(s) failed. And use the same more high-level message above - e.g. "Unable to classify data". That's all a connector/Flux user needs to know. A more advanced user can examine the stacktrace to debug it.